File: priceserver/modules/exchange/coinone.py
import requests
from datetime import datetime
from modules.settings import settings
from modules.moving_average import MovingAverage
from time import time

MOVING_AVG_SPAN = settings['UPDATER'].get("MOVING_AVG_SPAN", 30 * 60 * 1000)
COINONE_SERVER = "https://api.coinone.co.kr"

class coinone:
    id = "coinone"

    def fetch_markets(self):
        pairs = requests.get(COINONE_SERVER + "/ticker/?currency=_").json()

        markets = []
        for denom, data in pairs.items():
            markets.append({
                "id": denom,
                "symbol": f"{denom}/KRW".upper(),
                "base": denom,
                "quote": "KRW"
            })

        return markets

    def fetch_ticker(self, symbol: str):
        #denom = symbol.split("/")[0]
        #return requests.get(COINONE_SERVER + f"/ticker/?currency={denom}").json()
        result = requests.get('https://tb.coinone.co.kr/api/v1/chart/olhc/?site=coinoneluna&type=1m').json()
        ma = MovingAverage()

        for row in result['data']:
            if (time() * 1000 - int(row['DT'])) < MOVING_AVG_SPAN:
                ma.append((float(row['Low']) + float(row['High'])) / 2)

        return {
            "last": ma.get_price()
        }

File: priceserver-ts/src/provider/luna/Coinone.ts
import nodeFetch from 'node-fetch';
import { errorHandler } from 'lib/error';
import { num } from 'lib/num';
import * as logger from 'lib/logger';
import { Quoter, Trades } from '../base';

const candlestickUrl = {
  KRW: 'https://tb.coinone.co.kr/api/v1/chart/olhc/?site=coinoneluna&type=1m'
};

interface CandlestickResponse {
  success: boolean;
  data?: {
    DT: number;
    Open: string;
    Low: string;
    High: string;
    Close: string;
    Volume: string;
    Adj_Close: string;
  }[];
}

export class Coinone extends Quoter {
  private async fetchLatestTrades(quote: string): Promise<Trades> {
    // get latest candles
    const response: CandlestickResponse = await nodeFetch(candlestickUrl[quote], {
      timeout: this.options.timeout
    }).then(res => res.json());

    if (!response || !response.success || !Array.isArray(response.data) || response.data.length < 1) {
      logger.error(`${this.constructor.name}: wrong api response`, response ? JSON.stringify(response) : 'empty');
      throw 'skip';
    }

    return response.data
      .filter(row => parseFloat(row.Volume) > 0)
      .map(row => ({
        price: num(row.Close),
        volume: num(row.Volume),
        timestamp: +row.DT
      }));
  }

  protected async update(): Promise<boolean> {
    for (const quote of this.quotes) {
      // update last trades of LUNA/quote
      await this.fetchLatestTrades(quote)
        .then(trades => {
          if (!trades.length) {
            return;
          }

          this.setTrades(quote, trades);
          this.setPrice(quote, trades[trades.length - 1].price);
        })
        .catch(errorHandler);
    }

    return true;
  }
}

export default Coinone;
